test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs

```diff
@@ -237,202 +147,201 @@ public RunTimeTests(ITestOutputHelper output) : base(output)
                 {
-                    this.Logger.Log("Aborting listener WebSocket");
+                    TestUtility.Log("Aborting listener WebSocket");
                     cancelSource.Cancel();
                     await listenerStream.CloseAsync(cancelSource.Token);
                 }
 
                 byte[] readBuffer = new byte[1024];
                 await Assert.ThrowsAsync<RelayException>(() => clientStream.ReadAsync(readBuffer, 0, readBuffer.Length));
 
-                this.Logger.Log("Calling clientStream.Close");
+                TestUtility.Log("Calling clientStream.Close");
                 var clientCloseTask = clientStream.CloseAsync(CancellationToken.None);
-
-                this.Logger.Log("ListenerAbortWhileClientReading test end");
             }
             finally
             {
                 await this.SafeCloseAsync(listener);
             }
         }
 
-        [Fact]
-        public async Task NonExistantNamespace()
+        [Theory, DisplayTestMethodName]
+        [MemberData(nameof(AuthenticationTestPermutations))]
+        async Task NonExistantNamespaceTest(EndpointTestType endpointTestType)
         {
             HybridConnectionListener listener = null;
             try
             {
-                this.Logger.Log("NonExistantNamespace test start");
-
-                this.Logger.Log("Setting ConnectionStringBuilder.Endpoint to 'sb://fakeendpoint.com'");
+                TestUtility.Log("Setting ConnectionStringBuilder.Endpoint to 'sb://fakeendpoint.com'");
 
                 var fakeEndpointConnectionStringBuilder = new RelayConnectionStringBuilder(this.ConnectionString)
                 {
                     Endpoint = new Uri("sb://fakeendpoint.com")
                 };
+
+                if (endpointTestType == EndpointTestType.Authenticated)
+                {
+                    fakeEndpointConnectionStringBuilder.EntityPath = Constants.AuthenticatedEntityPath;
+                }
+                else
+                {
+                    fakeEndpointConnectionStringBuilder.EntityPath = Constants.UnauthenticatedEntityPath;
+                }
+
                 var fakeEndpointConnectionString = fakeEndpointConnectionStringBuilder.ToString();
 
                 listener = new HybridConnectionListener(fakeEndpointConnectionString);
                 var client = new HybridConnectionClient(fakeEndpointConnectionString);
 
-                await Assert.ThrowsAsync<RelayException>(() => listener.OpenAsync(TimeSpan.FromSeconds(30)));
+#if NET451
+                await Assert.ThrowsAsync<EndpointNotFoundException>(() => listener.OpenAsync());
+                await Assert.ThrowsAsync<EndpointNotFoundException>(() => client.CreateConnectionAsync());
+#else
+                await Assert.ThrowsAsync<RelayException>(() => listener.OpenAsync());
                 await Assert.ThrowsAsync<RelayException>(() => client.CreateConnectionAsync());
```

Comment:
Consider adding a "TODO" mentioning this is due to a bug in .NET Core (maybe even reference the issue#).